app/controllers/searches_controller.rb

```diff
@@ -3,7 +3,17 @@ class SearchesController < ApplicationController
 
   def show
     return unless params[:query] && params[:query].is_a?(String)
-    @gems = Rubygem.search(params[:query]).with_versions.paginate(page: @page)
+    if params[:es] == 'true'
+      begin
+        @gems = Rubygem.search(params[:query]).page(@page).records
+        @gems.size
```

Comment:
I need to trigger the call to ES in the controller rather than let it lazy load from the view. This way I can catch the failure and fallback.